### PR TITLE
Extend HCI with ECC.

### PR DESCRIPTION
**Pull request by LuigiGhionda**
This PR extends HCI with ECC up to the TCDM banks and add memory-mapped error-counting registers to collect statistics.

This PR supersedes and extends #65.

### CODE REVIEW

**Comment by yvantor** [comment]
> @LuigiGhionda is this ready for being merged?

(rebase apart)

File: Makefile [hunk 1]
```diff
@@ -59,7 +59,7 @@ endef
 ######################
 
 NONFREE_REMOTE ?= git@iis-git.ee.ethz.ch:pulp-restricted/pulp-cluster-nonfree.git
-NONFREE_COMMIT ?= ad58d90
+NONFREE_COMMIT ?= 06cb2203
```
Comment (yvantor) [inline Makefile:62]: Would you please also open a PR also on the nonfree if we need to update?

File: pulp-runtime
Comment (yvantor) [inline pulp-runtime:1]: Would you please also open PRs on the submodules?

Comment (yvantor) [reply pulp-runtime:1]: Edit: I saw this was already merged :)

**Review by yvantor — COMMENTED** [review]
Looks good, waiting for the working CI before approving!

**Review by yvantor — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (LuigiGhionda) [reply Makefile:62]: Done. See: https://iis-git.ee.ethz.ch/pulp-restricted/pulp-cluster-nonfree/-/merge_requests/11

**Comment by yvantor** [comment]
I bumped the nonfree to also include softex regression in the CI (I did not see a dedicated branch, maybe I missed it).
Merging as soon as the CI completes.

**Pull request merged**